Recipe Library (Oskar Nordin)

oskarnordin opened the pull request.

oskarnordin committed:
push 1

oskarnordin committed:
27/2

oskarnordin committed:
2025-03-06

oskarnordin committed:
Changed code to make cuisine filter stay when applying time filter

oskarnordin committed:
added random button that works

oskarnordin committed:
cleaning up code

oskarnordin committed:
cleaned up order of code + clearer comments

oskarnordin committed:
cleaned up css

oskarnordin committed:
cleaned up HTML with more readable indentation

oskarnordin committed:
Connecting recipe API to JavaScript code. Adjusting current code to work with it.

oskarnordin committed:
adjusted contianer for cards to fit 4 cards

oskarnordin committed:
added diet choice alternative

oskarnordin committed:
Removed cuisine option. Added diet option. Changed from checking array to checking boolean data in object.

oskarnordin committed:
added so that filter checks both boolean and array value regarding diet

oskarnordin committed:
Added search function

oskarnordin committed:
added loading animation and removed input display except search

oskarnordin committed:
Added local storage function

oskarnordin committed:
refactor the if-statement in getCurretnDietFiltered() function

oskarnordin committed:
add error icon

oskarnordin committed:
Sorted diet buttons alphabetically.

oskarnordin committed:
Added style to h3.

oskarnordin committed:
Removed semicolons for consistency.

oskarnordin committed:
add infinite scroll

VariaSlu commented on index.html:
```diff
@@ -0,0 +1,85 @@
+    
+
+    <script src="script.js"></script>
+</form>
```
There’s an unnecessary </form> closing tag at the very end of the file that doesn’t correspond to any opening <form> tag.

VariaSlu commented on index.html:
```diff
@@ -0,0 +1,85 @@
+            </label>
+        </div>
+    </div>
+    <div class="container">
```
If it's a placeholder for dynamically loaded recipes, consider adding a comment to clarify its purpose :)

VariaSlu commented on index.html:
```diff
@@ -0,0 +1,85 @@
+    <link rel="stylesheet" href="style.css">
+    <link rel="preconnect" href="https://fonts.googleapis.com">
+    <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
+    <link href="https://fonts.googleapis.com/css2?family=Kumbh+Sans:wght@100..900&display=swap" rel="stylesheet">
```
I like your choice of the custom font (Kumbh Sans), it is adding to the visual appeal for sure!

VariaSlu commented on script.js:
```diff
@@ -0,0 +1,211 @@
+    loadRecipes(filteredRecipes)
+}
+
+// Show loader. Hide after 2 seconds.
```
I've noticed the loader is hidden after 2 seconds using setTimeout, which may not align with the actual data loading time. This can cause inconsistent UX. It seems possible to dynamically hide the loader once the fetch completes
`showLoader();
fetch(URL)
    .then(() => hideLoader())
    .catch(() => hideLoader());
`

VariaSlu reviewed (commented):
Your CSS is well-structured and shows a lot of thoughtfulness, especially with the use of CSS variables like --primary-color and --secondary-color. This approach makes it easier to maintain and scale your design as the project grows. The layout is clean, and the responsiveness through media queries works well. Elements like .errorMessages and .errorButton not only look good but also feel intuitive for the user.

One thing to think about: if no recipes match a filter or search, the app doesn’t give the user any feedback. Maybe adding a simple message like “No recipes found. Try adjusting your search or filters!” would help guide the user and make the experience feel smoother. Overall, you’re doing a great job, and it’s clear you’re putting care into the details. Keep building on that!

HIPPIEKICK commented:
Can you share a link to your deployed page as well please 😇

oskarnordin commented:
https://recipe-libary-on.netlify.app/

On Wed, 26 Mar 2025 at 09:39, Matilda Brunemalm ***@***.***>
wrote:

> Can you share a link to your deployed page as well please 😇
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Technigo/js-project-recipe-library/pull/22#issuecomment-2753606404>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AJQPVURZWFBTEV7FYGAYFLD2WJKUJAVCNFSM6AAAAABZBSB2IOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDONJTGYYDMNBQGQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
> [image: HIPPIEKICK]*HIPPIEKICK* left a comment
> (Technigo/js-project-recipe-library#22)
> <https://github.com/Technigo/js-project-recipe-library/pull/22#issuecomment-2753606404>
>
> Can you share a link to your deployed page as well please 😇
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Technigo/js-project-recipe-library/pull/22#issuecomment-2753606404>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AJQPVURZWFBTEV7FYGAYFLD2WJKUJAVCNFSM6AAAAABZBSB2IOVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDONJTGYYDMNBQGQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

HIPPIEKICK reviewed (changes requested):
Good job with this project Oskar! Although I have some things for you to work on, I think you should be proud of what you’ve accomplished. First, I’d like to ask you to have another look at how your page looks on mobile. It doesn’t really fit. An idea would be to decrease the padding a bit to let the content take a little bit more width (we don’t have so much width to work on on mobile to start with 😅)

Then I would also like you to clean up your code:
- Be consistent with whether or not you’re using semicolons.
- Remove console.log from production code.
- Set your VSC tab to 2 spaces. It’s so hard to read with 4 spaces 🙈

And I noticed that you completed some of the stretch goals (great job!) but your loader is not showing as expected, so please have a look at that as well if you’re aiming for VG.

oskarnordin commented:
Thanks for the feedback. I'm working on fixing the suggestions you
mentioned. Didn't even notice the 4-space indentation. 💀💀💀 Will also
check the loading stretch goal.

On Thu, 27 Mar 2025 at 15:22, Matilda Brunemalm ***@***.***>
wrote:

> ***@***.**** requested changes on this pull request.
>
> Good job with this project Oskar! Although I have some things for you to
> work on, I think you should be proud of what you’ve accomplished. First,
> I’d like to ask you to have another look at how your page looks on mobile.
> It doesn’t really fit. An idea would be to decrease the padding a bit to
> let the content take a little bit more width (we don’t have so much width
> to work on on mobile to start with 😅)
>
> Then I would also like you to clean up your code:
>
>    - Be consistent with whether or not you’re using semicolons.
>    - Remove console.log from production code.
>    - Set your VSC tab to 2 spaces. It’s so hard to read with 4 spaces 🙈
>
> And I noticed that you completed some of the stretch goals (great job!)
> but your loader is not showing as expected, so please have a look at that
> as well if you’re aiming for VG.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Technigo/js-project-recipe-library/pull/22#pullrequestreview-2721883646>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AJQPVUWJHW6LS2WILFM3MOT2WP3QVAVCNFSM6AAAAABZBSB2IOVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDOMRRHA4DGNRUGY>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

oskarnordin committed:
Switched from 4 to 2 indentations. Removed Console.logs's and made the loading feature work.

oskarnordin commented:
- Be consistent with whether or not you’re using semicolons.
   Removed them.
   - Remove console.log from production code. Removed them.
   - Set your VSC tab to 2 spaces. It’s so hard to read with 4 spaces Set
   it to 2 spaces.
   - + Made the loading feature work.

https://recipe-libary-on.netlify.app/

// Oskar

On Thu, 27 Mar 2025 at 17:03, Oskar Nordin ***@***.***> wrote:

> Thanks for the feedback. I'm working on fixing the suggestions you
> mentioned. Didn't even notice the 4-space indentation. 💀💀💀 Will also
> check the loading stretch goal.
>
> On Thu, 27 Mar 2025 at 15:22, Matilda Brunemalm ***@***.***>
> wrote:
>
>> ***@***.**** requested changes on this pull request.
>>
>> Good job with this project Oskar! Although I have some things for you to
>> work on, I think you should be proud of what you’ve accomplished. First,
>> I’d like to ask you to have another look at how your page looks on mobile.
>> It doesn’t really fit. An idea would be to decrease the padding a bit to
>> let the content take a little bit more width (we don’t have so much width
>> to work on on mobile to start with 😅)
>>
>> Then I would also like you to clean up your code:
>>
>>    - Be consistent with whether or not you’re using semicolons.
>>    - Remove console.log from production code.
>>    - Set your VSC tab to 2 spaces. It’s so hard to read with 4 spaces 🙈
>>
>> And I noticed that you completed some of the stretch goals (great job!)
>> but your loader is not showing as expected, so please have a look at that
>> as well if you’re aiming for VG.
>>
>> —
>> Reply to this email directly, view it on GitHub
>> <https://github.com/Technigo/js-project-recipe-library/pull/22#pullrequestreview-2721883646>,
>> or unsubscribe
>> <https://github.com/notifications/unsubscribe-auth/AJQPVUWJHW6LS2WILFM3MOT2WP3QVAVCNFSM6AAAAABZBSB2IOVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDOMRRHA4DGNRUGY>
>> .
>> You are receiving this because you authored the thread.Message ID:
>> ***@***.***>
>>
>

oskarnordin committed:
Update

oskarnordin committed:
Merge branch 'main' of https://github.com/oskarnordin/js-project-recipe-library
